### ISSUE TITLE
Improve error messages when using `RMatrix`

### ISSUE THREAD

**Issue by daniellga** [extendr/extendr#456]
When calling a function with a different kind of `RMatrix` the error states that the input given is not an RMatrix, which can be misleading. I think the error could be a bit more specific. Maybe something like `Expected a matrix of doubles` or even `Expected another type of matrix`.

```
rextendr::rust_function("fn take_rfloat_matrix(rmatrix: RMatrix<Rfloat>) { println!(\"ok\"); }")
rextendr::rust_function("fn take_rcplx_matrix(rmatrix: RMatrix<Rcplx>) { println!(\"ok\"); }")
floatmatrix = matrix(c(1,2,3,4))
complexmatrix = matrix(c(1+1i, 2+2i, 3+3i, 4+4i))
is.matrix(floatmatrix)
is.matrix(complexmatrix)
take_rfloat_matrix(floatmatrix)
take_rcplx_matrix(complexmatrix)
take_rfloat_matrix(complexmatrix)
take_rcplx_matrix(floatmatrix)
```
```
> floatmatrix = matrix(c(1,2,3,4))
> complexmatrix = matrix(c(1+1i, 2+2i, 3+3i, 4+4i))
> is.matrix(floatmatrix)
[1] TRUE
> is.matrix(complexmatrix)
[1] TRUE
> take_rfloat_matrix(floatmatrix)
ok
> take_rcplx_matrix(complexmatrix)
ok
> take_rfloat_matrix(complexmatrix)
Error in take_rfloat_matrix(complexmatrix) : Expected a matrix.
> take_rcplx_matrix(floatmatrix)
Error in take_rcplx_matrix(floatmatrix) : Expected a matrix.
```

**Comment by Ilia-Kosenkov** [comment]
It is a somewhat known issue, but we cannot really fix it before #336. 
You might get better error info if you decorate functions with `#[extendr(use_try_from = true)]`

**Comment by daniellga** [comment]
Oh it gives a better error indeed. Gonna close this one then.
